src/openforms/logging/service.py

```diff
@@ -81,34 +82,40 @@ def collect_failed_emails(since: datetime) -> Iterable[FailedEmail]:
 
 def collect_failed_registrations(
     since: datetime,
-) -> Iterable[dict[str, FailedRegistration]]:
+) -> list[FailedRegistration]:
     logs = TimelineLogProxy.objects.filter(
         timestamp__gt=since,
         extra_data__log_event="registration_failure",
     ).order_by("timestamp")
 
-    if not logs:
-        return []
-
     form_sorted_logs = sorted(logs, key=lambda x: x.content_object.form.admin_name)
 
     grouped_logs = groupby(form_sorted_logs, key=lambda log: log.content_object.form)
 
-    failed_registrations = {}
+    failed_registrations = []
```

Comment:
Changed this to a list in order to have the same structure in all the similar functions.